Can't build with fresh checkout

**Version**
52b27faa09715b5835804de7abf6998e028736fc

**Platform**
x86_64 Linux

**Description**
Trying to run `cargo test` after cloning this repo results in the following errors:

```
error[E0432]: unresolved import `super::super::Compat`
   --> src/common/io/rewind.rs:111:9
    |
111 |     use super::super::Compat;
    |         ^^^^^^^^^^^^^^^^^^^^ no `Compat` in `common::io`
    |
note: found an item that was configured out
   --> src/common/io/mod.rs:6:30
    |
6   | pub(crate) use self::compat::Compat;
    |                              ^^^^^^

error[E0412]: cannot find type `Compat` in module `crate::common::io`
   --> src/upgrade.rs:357:44
    |
357 |             .downcast::<crate::common::io::Compat<std::io::Cursor<Vec<u8>>>>()
    |                                            ^^^^^^ not found in `crate::common::io`
    |
note: found an item that was configured out
   --> src/common/io/mod.rs:6:30
    |
6   | pub(crate) use self::compat::Compat;
    |                              ^^^^^^

error[E0599]: no function or associated item named `new` found for struct `upgrade::Upgraded` in the current scope
   --> src/upgrade.rs:354:34
    |
65  | pub struct Upgraded {
    | ------------------- function or associated item `new` not found for this struct
...
354 |         let upgraded = Upgraded::new(Mock, Bytes::new());
    |                                  ^^^ function or associated item not found in `Upgraded`

warning: unused import: `tokio::io::AsyncReadExt`
   --> src/common/io/rewind.rs:114:9
    |
114 |     use tokio::io::AsyncReadExt;
    |         ^^^^^^^^^^^^^^^^^^^^^^^
    |
    = note: `#[warn(unused_imports)]` on by default

Some errors have detailed explanations: E0412, E0432, E0599.
For more information about an error, try `rustc --explain E0412`.
warning: `hyper` (lib test) generated 1 warning
error: could not compile `hyper` (lib test) due to 3 previous errors; 1 warning emitted
```

--- a/src/common/io/rewind.rs
+++ b/src/common/io/rewind.rs
@@ -106,6 +106,10 @@ where
     }
 }
 
+#[cfg(all(
+    any(feature = "client", feature = "server"),
+    any(feature = "http1", feature = "http2"),
+))]
 #[cfg(test)]
 mod tests {
     use super::super::Compat;

--- a/src/upgrade.rs
+++ b/src/upgrade.rs
@@ -345,6 +345,10 @@ mod sealed {
     }
 }
 
+#[cfg(all(
+    any(feature = "client", feature = "server"),
+    any(feature = "http1", feature = "http2"),
+))]
 #[cfg(test)]
 mod tests {
     use super::*;

